app/helpers/requests_helper.rb

```diff
@@ -95,4 +95,13 @@ def requester_info(user)
       user.library_id
     end
   end
+
+  # For the reading room information, we need to check if 'ARS' is in the location details
+  # for the library. An example is SAL3, which should show the ARS reading room information
+  # and so should return ARS as the library code for the reading room text block.
+  # This logic will be extended in the future to cover any location that has a pageAeonSite value.
+  def aeon_reading_room_code
+    details = Folio::Types.locations.find_by(code: current_request.origin_location).details
+    details.key?('pageAeonSite') && details['pageAeonSite'] == 'ARS' ? 'ARS' : current_request.origin_library_code
+  end
```

Comment:
Is there a reason we can't move this functionality into https://github.com/sul-dlss/sul-requests/blob/8b883d5d15b9698a553cf65ed3d5bf784c6b2899/app/models/request.rb#L178

Reply:
For the origin library code function that is more generally applicable, I am not sure this mapping (i.e. from SAL3-PAGE-AS to ARS) applies to any other part of the application except for the reading room page.  In other words, for other situations where you would want an origin library code, this mapping doesn't apply.  I suppose we could do an aeon reading room code within the request class instead, but it may not matter for the moment because we'll be rewriting that anyway.